src/Microsoft.DotNet.Build.Tasks.Feed/CopyBlobDirectory.cs

```diff
@@ -0,0 +1,138 @@
+                    // Grab the source blob path from the source info and combine with the target blob path.
+                    BlobUrlInfo sourceBlobInfo = new BlobUrlInfo(sourceUri);
+
+                    // For now the source and target storage accounts should be the same, so the same accont key is used for each.
+                    if (sourceBlobInfo.AccountName != targetUrlInfo.AccountName)
+                    {
+                        Log.LogError($"Source and target storage accounts should be identical");
+                    }
+                    else
+                    {
+                        CloudBlobContainer sourceContainer = client.GetContainerReference(sourceBlobInfo.ContainerName);
+
+                        Log.LogMessage(MessageImportance.Normal, $"Listing blobs in {sourceUri}");
+
+                        // Get all source URI's with the blob prefix
+                        BlobContinuationToken token = null;
+                        List<IListBlobItem> sourceBlobs = new List<IListBlobItem>();
+                        do
+                        {
+                            BlobResultSegment segment = await sourceContainer.ListBlobsSegmentedAsync(sourceBlobInfo.BlobPath, true,
+                                BlobListingDetails.None, null, token, new BlobRequestOptions(), null);
+                            token = segment.ContinuationToken;
+                            sourceBlobs.AddRange(segment.Results);
+                        }
+                        while (token != null);
+
+                        // Ensure the source exists
+                        if (!SkipIfMissing && sourceBlobs.Count == 0)
+                        {
+                            Log.LogError($"No blobs found in {sourceUri}");
+                        }
+
+                        await Task.WhenAll(sourceBlobs.Select(async blob =>
+                        {
+                            // Determine the relative URI for the target.  This works properly when the
+                            // trailing slash is left off of the source and target URIs.
+                            string relativeBlobPath = blob.Uri.ToString().Substring(sourceUri.Length);
+                            string specificTargetUri = GetCanonicalStorageUri(targetUri, relativeBlobPath);
+                            BlobUrlInfo specificTargetBlobUrlInfo = new BlobUrlInfo(specificTargetUri);
+                            CloudBlob targetBlob = targetContainer.GetBlobReference(specificTargetBlobUrlInfo.BlobPath);
+
+                            Log.LogMessage(MessageImportance.Normal, $"Merging {blob.Uri.ToString()} into {targetBlob.Uri.ToString()}");
+
+                            if (!Overwrite && await targetBlob.ExistsAsync())
+                            {
+                                Log.LogError($"Target blob {targetBlob.Uri.ToString()} already exists.");
+                            }
+
+                            await targetBlob.DeleteIfExistsAsync();
```

Comment:
Do we have to do an explicit delete? I would hope there was an option you could pass to Copy. Doing the explicit delete on overwrites adds a window of time for race conditions.

Reply:
It overwrite's by default, so  this has been removed.